src/scripts/main.js

```diff
@@ -1,3 +1,227 @@
 'use strict';
 
-// write code here
+function updateSortDirection(columnIndex) {
+  if (columnIndex === lastSortedColumn) {
+    sortDirection = -sortDirection;
+  } else {
+    sortDirection = 1;
+  }
+
+  lastSortedColumn = columnIndex;
+}
+
+function getCellValue(row, columnIndex) {
+  return currencyToNumber(row.cells[columnIndex].innerText);
+}
+
+function currencyToNumber(value) {
+  if (value[0] === '$') {
+    return +value.slice(1).split(',').join('');
+  }
+
+  return isNaN(value) ? value : +value;
+}
+
+function sortTable(columnIndex) {
+  const rows = [...tbody.querySelectorAll('tr')];
+
+  rows.sort((row1, row2) => {
+    const value1 = getCellValue(row1, columnIndex);
+    const value2 = getCellValue(row2, columnIndex);
+
+    return typeof value1 === 'number'
+      ? (value1 - value2) * sortDirection
+      : value1.toString().localeCompare(value2.toString()) * sortDirection;
+  });
+
+  table.tBodies[0].append(...rows);
+}
+
+function createInputWithLabel(id, text, type = 'text', dataQa = id) {
+  const label = document.createElement('label');
+
+  label.innerHTML = `<span>${text}</span><input id="${id}" type="${type}" data-qa="${dataQa}">`;
+
+  return label;
+}
+
+function pushNotification(title, description, type) {
+  const div = document.createElement('div');
+
+  div.classList.add('notification', type);
+
+  div.innerHTML = `<h2 class="title">${title}</h2><p>${description}</p>`;
+  document.body.append(div);
+
+  setTimeout(() => div.remove(), 2000);
+}
+
+const table = document.querySelector('table');
+const headers = table.querySelectorAll('th');
+const tbody = table.querySelector('tbody');
+let sortDirection = 1;
+let lastSortedColumn = -1;
+let editingCell = null;
+
+table.querySelector('thead').addEventListener('click', (e) => {
+  const header = e.target.closest('th');
+
+  if (!header) {
+    return;
+  }
+
+  const index = [...headers].indexOf(header);
+
+  updateSortDirection(index);
+
+  sortTable(index);
+});
+
+tbody.addEventListener('click', (e) => {
+  const row = e.target.closest('tr');
+
+  if (!row) {
+    return;
+  }
+
+  tbody.querySelectorAll('tr').forEach((r) => r.classList.remove('active'));
+  row.classList.add('active');
+});
+
+tbody.addEventListener('dblclick', (e) => {
+  const cell = e.target.closest('td');
+
+  if (!cell || editingCell) {
+    return;
+  }
+
+  editingCell = cell;
+
+  const initialValue = cell.innerText;
+
+  cell.innerHTML = '';
+
+  const input = document.createElement('input');
+
+  input.classList.add('cell-input');
+  input.value = '';
```

Comment:
The `input.value` is set to an empty string here, which might not be the intended behavior. Consider setting it to `initialValue` to allow users to see and edit the current cell value.